### PR TITLE
[Doc] Add documentation for `fetchJson`

### PR DESCRIPTION
**Pull request by slax57**

### CODE REVIEW

File: docs/DataProviders.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ In this example, the `simpleRestProvider` accepts a second parameter to set auth
 
 The `dataProvider` doesn't "speak" HTTP, so it doesn't have the notion of HTTP headers. If you need to pass custom headers to the API, the syntax depends on the Data Provider you use. 
 
-For instance, the `simpleRestProvider` function accepts an HTTP client function as second argument. By default, it uses react-admin's `fetchUtils.fetchJson()` function as HTTP client. It's similar to HTML5 `fetch()`, except it handles JSON decoding and HTTP error codes automatically.
+For instance, the `simpleRestProvider` function accepts an HTTP client function as second argument. By default, it uses react-admin's [`fetchUtils.fetchJson()`](./fetchJson.md) function as HTTP client. It's similar to HTML5 `fetch()`, except it handles JSON decoding and HTTP error codes automatically.
```
Comment (djhi) [inline docs/DataProviders.md:179]:
```suggestion
For instance, the `simpleRestProvider` function accepts an HTTP client function as its second argument. By default, it uses react-admin's [`fetchUtils.fetchJson()`](./fetchJson.md) function as the HTTP client. It's similar to the HTML5 `fetch()`, except it handles JSON decoding and HTTP error codes automatically.
```

File: docs/fetchJson.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
+---
+layout: default
+title: "fetchJson"
+---
+
+# `fetchJson`
+
+React-admin includes a `fetchJson` utility function to make HTTP calls. It's a wrapper around the browser's `fetch` function, which adds the following features:
```
Comment (djhi) [inline docs/fetchJson.md:8]:
```suggestion
React-admin includes a `fetchJson` utility function to make HTTP calls. It's a wrapper around the browser's `fetch` function, that adds the following features:
```

File: docs/fetchJson.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+-   It handles the JSON decoding of the response
+-   It handles HTTP errors codes by throwing an `HttpError`
+
+## Usage
+
+You can use it to make HTTP calls directly, to build a custom [`dataProvider`](./DataProviderIntroduction.md), or pass it directly to any `dataProvider` that supports it, such as [`ra-data-simple-rest`](https://github.com/marmelab/react-admin/tree/master/packages/ra-data-simple-rest).
+
+```jsx
+import { fetchUtils, Admin, Resource } from 'react-admin';
+import simpleRestProvider from 'ra-data-simple-rest';
```
Comment (djhi) [inline docs/fetchJson.md:22]:
```suggestion
import simpleRestProvider from 'ra-data-simple-rest';
import { PostList } from './posts';
```

File: docs/fetchJson.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+const dataProvider = simpleRestProvider('http://path.to.my.api/', httpClient);
+
+const App = () => (
+    <Admin dataProvider={dataProvider}>
+        <Resource name="posts" list={PostList} />
+    </Admin>
+);
+```
+
+**Tip:** `fetchJson` is included in the `fetchUtils` member of the `react-admin` package.
```
Comment (djhi) [inline docs/fetchJson.md:38]:
```suggestion
**Tip:** `fetchJson` is included in the `fetchUtils` object exported by the `react-admin` package.
```

File: docs/fetchJson.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+-   `body` **string** The response body
+-   `json` **Object** The response body, parsed as JSON
+
+## Adding Custom Headers
+
+Here is an example of how to add custom headers to all requests:
+
+```jsx
+import { fetchUtils, Admin, Resource } from 'react-admin';
+import simpleRestProvider from 'ra-data-simple-rest';
```
Comment (djhi) [inline docs/fetchJson.md:66]:
```suggestion
import simpleRestProvider from 'ra-data-simple-rest';
import { PostList } from './posts';
```

File: packages/ra-core/src/dataProvider/fetch.ts [hunk 6]
```diff
@@ -27,6 +27,22 @@ export const createHeadersFromOptions = (options: Options): Headers => {
     return requestHeaders;
 };
 
+/**
+ * Utility function to make HTTP calls. It's similar to HTML5 `fetch()`, except it handles JSON decoding and HTTP error codes automatically.
```
Comment (djhi) [inline packages/ra-core/src/dataProvider/fetch.ts:31]:
```suggestion
 * Utility function to make HTTP calls. It's similar to the HTML5 `fetch()`, except it handles JSON decoding and HTTP error codes automatically.
```

**Review by djhi — CHANGES_REQUESTED** [review]

File: docs/fetchJson.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+const App = () => (
+    <Admin dataProvider={dataProvider}>
+        <Resource name="posts" list={PostList} />
+    </Admin>
+);
+```
+
+**Tip:** `fetchJson` is included in the `fetchUtils` member of the `react-admin` package.
+
+### Parameters
```
Comment (fzaninotto) [inline docs/fetchJson.md:40]: We don't need 2 levels of heading here. Make it a top-level heading (`###` => `##`)

**Review by djhi — APPROVED** [review]

**Pull request merged**